tests/phpunit/integration/api/compiler/beans-compiler-options/flush.php

```diff
@@ -0,0 +1,41 @@
+<?php
+/**
+ * Tests the flush() method of _Beans_Compiler_Options.
+ *
+ * @package Beans\Framework\Tests\Integration\API\Compiler
+ *
+ * @since   1.5.0
+ */
+
+namespace Beans\Framework\Tests\Integration\API\Compiler;
+
+use _Beans_Compiler_Options;
+use Beans\Framework\Tests\Integration\API\Compiler\Includes\Compiler_Options_Test_Case;
+use Brain\Monkey;
+use org\bovigo\vfs\vfsStream;
+
+require_once dirname( __DIR__ ) . '/includes/class-compiler-options-test-case.php';
+
+/**
+ * Class Tests_BeansCompilerOptions_Flush
+ *
+ * @package Beans\Framework\Tests\Integration\API\Compiler
+ * @group   api
+ * @group   api-compiler
+ */
+class Tests_BeansCompilerOptions_Flush extends Compiler_Options_Test_Case {
+
+	/**
+	 * Test _Beans_Compiler_Options::flush() should should not remove the cached directory when this is not a 'compiler
+	 * cache flush'.
+	 */
+	public function test_should_not_remove_cached_dir_when_not_a_flush() {
+		// Check that the cached directory exists before we start.
+		$this->directoryExists( vfsStream::url( 'compiled/beans/compiler/' ) );
+
+		$this->assertNull( ( new _Beans_Compiler_Options() )->flush() );
+
+		// Check that it still exists and was not removed.
+		$this->directoryExists( vfsStream::url( 'compiled/beans/compiler/' ) );
+	}
+}
```

Comment:
Let's have a second test that proves the opposite case -- "should remove cached dir when is a flush".

Reply:
Whew, it took a bit of doing.  But all of us will be glad here.  

In order to test that the cached directory was removed (flushed), we needed a way to prevent `wp_normalize_path` from running. Why?

The virtual filesystem uses a prefix of `vfs:://` to recognize its unique path structure.  Running it through `wp_normalize_path` strips the second `/`.  That causes our virtual filesystem to not work.

The solution here is to mock it using Patchwork.  But in the integration test suite, WordPress and Beans loaded before our Test Cases.  Hmm, how do we fix this?

Brain Monkey has a loader for Patchwork.  I added that loader to the integration's bootstrap file.  We now load Patchwork _before_ WordPress and Beans. 

What does that give us?

We can now mock any WordPress and Beans functions in our integration suite without getting a fatal error for "too early."